batcher_test.go

```diff
@@ -1,5 +1,7 @@
 package batcher_test
 
+// NOTE: please review unit-tests
+
```

Comment:
`Operation.go` has two different operations that enable batching (`WithBatching()` and `AllowBatch()`). 

A quick search for `WithBatching()` returns lines 147, 238, and 282 in this file. The alternative operation, `AllowBatch()`, is only used in the sample. Not seeing a real difference between the two.

Reply:
There is no functional difference between the two, they serve the same purpose. However, AllowBatch() sets to TRUE while WithBatching() takes a bool. In normal usage, I would expect someone to know what the operation is and just use AllowBatch() if appropriate. However, in cases where there the appropriate behavior of the query isn't known until runtime it might be easier to set WithBatching() to a variable that describes whether or not it can be batched. It is just a convenience function.